site/content/faq/500-what-about-typescript-support.md

```diff
@@ -10,3 +10,9 @@ To declare the type of a reactive variable in a Svelte template, you should use
 let x: number;
 $: x = count + 1;
 ```
+
+To `import` a type or interface make sure to use [TypeScript's `type` modifier](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export):
+
+```
+import type { SomeInterface } from './SomeFile';
+```
```

Comment:
```suggestion
To `import` a type or interface make sure to use [TypeScript's `type` modifier](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export):

```ts
import type { SomeInterface } from './SomeFile';
```

This is because svelte-preprocess doesn't know whether an import is a type or a value, and can't safely erase imports which only contain types.
```

There we go. also gives the reasoning.

Reply:
I'm having a bit of a hard time parsing the last sentence. Would the following be equivalent?

> This is necessary to allow svelte-preprocess to distinguish whether an import is a type or a value, so that it can safely erase the type imports.

Though, I'm not sure it fully explains the reasoning. Why does svelte-preprocess need to do anything special to erase the types and why isn't this an issue when you normally call the TypeScript compiler?

Reply:
The TS compiler method which compiles whole projects knows all files and its dependencies and therefore knows which imports are types and which not. `svelte-preprocess` uses the "transpile this code given without any outside knowledge" method of the TS compiler.

Reply:
Ok. What if we say something like:

> Because `svelte-preprocess` compiles one file at a time, make sure to use Typescript's `type` modifier when importing a type or interface:

Reply:
IMO this is no real explanation anymore. I also think it's good to have "do this" and "why you have to do this" in two separate sentences (as it's now with Orta's suggestion).

So my suggestion would be to leave your first part as is and instead fine-tune the explanation.

> This is because `svelte-preprocess` doesn't know whether an import is a type or a value: It only transpiles one file at a time without knowledge of the other files and therefore can't safely erase imports which only contain types.

Reply:
Ok. Updated